Segmentation Fault in tokenTransport.RoundTrip When Access Token Missing

### Description
The MongoDB Atlas CLI crashes with a segmentation fault when the access token is missing but a valid refresh token exists. The crash occurs in the `tokenTransport.RoundTrip` method in the `atlas-cli-core` dependency, preventing any CLI operations that require authentication.

### Steps to Reproduce
1. Have a valid OAuth session (both access and refresh tokens present)
2. Delete the access token:
   ```bash
   security delete-generic-password -s "atlascli_default" -a "access_token"
   ```
3. Run any CLI command that requires authentication:
   ```bash
   atlas whoami
   # or
   atlas projects list
   ```

### Expected Behavior
The CLI should automatically use the refresh token to obtain a new access token and continue operation without user intervention.

### Actual Behavior
Segmentation fault occurs with the following stack trace:
```
Logged in as  account
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x10 pc=0x1055e7680]

goroutine 1 [running]:
github.com/mongodb/atlas-cli-core/transport.(*tokenTransport).RoundTrip(0x14000812180, 0x1400047c500)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/mongodb/atlas-cli-core@v0.0.0-20250909113945-ffb322e05f59/transport/transport.go:101 +0x60
net/http.send(0x1400047c500, {0x1068d9540, 0x14000812180}, {0x104ee6218?, 0x8?, 0x0?})
	/opt/homebrew/Cellar/go/1.25.3/libexec/src/net/http/client.go:259 +0x478
net/http.(*Client).send(0x140008122a0, 0x1400047c500, {0x107b3ed80?, 0x10c654aa8?, 0x0?})
	/opt/homebrew/Cellar/go/1.25.3/libexec/src/net/http/client.go:180 +0x88
net/http.(*Client).do(0x140008122a0, 0x1400047c500)
	/opt/homebrew/Cellar/go/1.25.3/libexec/src/net/http/client.go:729 +0x710
net/http.(*Client).Do(...)
	/opt/homebrew/Cellar/go/1.25.3/libexec/src/net/http/client.go:587
go.mongodb.org/atlas/mongodbatlas.DoRequestWithClient({0x1068edaf0, 0x1400075aff0}, 0x140008122a0, 0x1400047c3c0)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/go.mongodb.org/atlas@v0.38.0/mongodbatlas/mongodbatlas.go:585 +0xb8
go.mongodb.org/atlas/mongodbatlas.(*Client).Do(0x140000ff688, {0x1068edaf0, 0x1400075aff0}, 0x1400047c3c0, {0x0, 0x0})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/go.mongodb.org/atlas@v0.38.0/mongodbatlas/mongodbatlas.go:465 +0x60
github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store.(*Store).SendEvents(0x14000062780, {0x10637f5a0?, 0x14000010948?})
	/private/tmp/mongodb-atlas-cli-20251023-7587-km0z3c/mongodb-atlas-cli-atlascli-v1.50.0/internal/store/telemetry.go:31 +0xc0
github.com/mongodb/mongodb-atlas-cli/atlascli/internal/telemetry.(*tracker).trackCommand(0x1400045c280, {{0x0?, 0x0?}, {0x0?, 0x14000859ab8?}}, {0x0, 0x0, 0x14000a2c740?})
	/private/tmp/mongodb-atlas-cli-20251023-7587-km0z3c/mongodb-atlas-cli-atlascli-v1.50.0/internal/telemetry/tracker.go:145 +0x4b8
github.com/mongodb/mongodb-atlas-cli/atlascli/internal/telemetry.FinishTrackingCommand({{0x0?, 0x0?}, {0x0?, 0x140004488b0?}})
	/private/tmp/mongodb-atlas-cli-20251023-7587-km0z3c/mongodb-atlas-cli-atlascli-v1.50.0/internal/telemetry/command.go:56 +0x78
github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/root.Builder.func2(0x140002ad208, {0x107b3c4a0?, 0x0?, 0x0?})
	/private/tmp/mongodb-atlas-cli-20251023-7587-km0z3c/mongodb-atlas-cli-atlascli-v1.50.0/internal/cli/root/builder.go:161 +0x160
github.com/spf13/cobra.(*Command).execute(0x140002ad208, {0x107b3c4a0, 0x0, 0x0})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1037 +0x8d8
github.com/spf13/cobra.(*Command).ExecuteC(0x140002ac908)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).ExecuteContextC(...)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1080
main.execute({0x1068edaf0, 0x1400075aff0}, 0x140002ac908)
	/private/tmp/mongodb-atlas-cli-20251023-7587-km0z3c/mongodb-atlas-cli-atlascli-v1.50.0/cmd/atlas/atlas.go:40 +0x98
main.main()
	/private/tmp/mongodb-atlas-cli-20251023-7587-km0z3c/mongodb-atlas-cli-atlascli-v1.50.0/cmd/atlas/atlas.go:103 +0xd0
```

### Root Cause
Nil pointer dereference in `github.com/mongodb/atlas-cli-core/transport/transport.go:101` in the `tokenTransport.RoundTrip` method when attempting to make authenticated HTTP requests without a valid access token.

### Impact
- **User Experience**: CLI becomes completely unusable for all authenticated commands
- **Reliability**: Broken automatic token recovery mechanism
- **Security**: OAuth refresh flow is non-functional, requiring manual re-authentication
- **All Commands Affected**: Any command requiring authentication will crash

### Environment
- **CLI Version**: 1.50.0
- **OS**: macOS (Silicon)
- **Authentication**: OAuth
- **Keychain**: macOS Keychain

### Additional Context
The refresh token remains intact in the keychain during this failure. The issue specifically affects the HTTP transport layer's ability to handle missing access tokens gracefully.

--- a/build/package/purls.txt
+++ b/build/package/purls.txt
@@ -73,7 +73,7 @@ pkg:golang/github.com/mholt/archives@v0.1.5
 pkg:golang/github.com/mikelolasagasti/xz@v1.0.1
 pkg:golang/github.com/minio/minlz@v1.0.1
 pkg:golang/github.com/mongodb-forks/digest@v1.1.0
-pkg:golang/github.com/mongodb/atlas-cli-core@v0.0.0-20250909113945-ffb322e05f59
+pkg:golang/github.com/mongodb/atlas-cli-core@v0.0.0-20251029131602-d10cd0d198ab
 pkg:golang/github.com/montanaflynn/stats@v0.7.1
 pkg:golang/github.com/nwaples/rardecode/v2@v2.2.0
 pkg:golang/github.com/pelletier/go-toml/v2@v2.2.4

--- a/go.mod
+++ b/go.mod
@@ -29,7 +29,7 @@ require (
 	github.com/mattn/go-isatty v0.0.20
 	github.com/mholt/archives v0.1.5
 	github.com/mongodb-labs/cobra2snooty v1.19.1
-	github.com/mongodb/atlas-cli-core v0.0.0-20250909113945-ffb322e05f59
+	github.com/mongodb/atlas-cli-core v0.0.0-20251029131602-d10cd0d198ab
 	github.com/pelletier/go-toml v1.9.5
 	github.com/pkg/browser v0.0.0-20240102092130-5ac0b6a4141c
 	github.com/shirou/gopsutil/v4 v4.25.9

--- a/internal/cli/auth/whoami.go
+++ b/internal/cli/auth/whoami.go
@@ -46,6 +46,9 @@ func authTypeAndSubject() (string, string, error) {
 		return "service account", config.ClientID(), nil
 	case config.UserAccount:
 		subject, _ := config.AccessTokenSubject()
+		if subject == "" {
+			return "", "", ErrUnauthenticated
+		}
 		return "account", subject, nil
 	case config.NoAuth:
 		return "", "", ErrUnauthenticated

